File: src/main/java/grp/faq/dao/QuestionMapper.java
package grp.faq.dao;

import grp.faq.entity.Question;

public interface QuestionMapper {
    int deleteByPrimaryKey(Long cQid);

    int insert(Question record);

    int insertSelective(Question record);

    Question selectByPrimaryKey(Long cQid);

    int updateByPrimaryKeySelective(Question record);

    int updateByPrimaryKey(Question record);
}

File: src/main/scala/grp/faq/dao/QuestionMapper.scala
package grp.faq.dao

import java.util.Map
import java.util.List

import com.alibaba.fastjson.JSONObject
import grp.faq.entity.Question
import org.apache.ibatis.annotations.Param

trait QuestionMapper {
  def getQUestionByUsername(@Param("username")username: String, @Param("index")index: Int): List[Question]

  def getQuestionNumber(): Int

  def getQuestion(question: Question): Map[Object,Object]

  def addQuestion(question: Question): Int

  def getQuestionRank(@Param("index")index: Int): List[Question]

  def getTodayQuestion(): Long
}
